pkg/router/functionHandler.go

```diff
@@ -220,13 +221,19 @@ func (roundTripper *RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Re
 			req.URL.Scheme = roundTripper.serviceURL.Scheme
 			req.URL.Host = roundTripper.serviceURL.Host
 
-			// To keep the function run container simple, it
-			// doesn't do any routing.  In the future if we have
-			// multiple functions per container, we could use the
-			// function metadata here.
-			// leave the query string intact (req.URL.RawQuery)
-			req.URL.Path = "/"
-
+			if strings.HasSuffix(roundTripper.funcHandler.httpTrigger.Spec.RelativeURL, "/*") {
```

Comment:
We need to do something similar to this over here: https://www.getambassador.io/docs/edge-stack/latest/topics/using/rewrites/#rewrites